static/canvas.js

```diff
@@ -126,8 +128,27 @@ class Canvas {
         /* Event for moving the mouse. Primarily used for enlarging a box and moving it. */
         canvas.addEventListener('mousemove', function (e) {
             var mouse = myState.getMouse(e);
-            if (!myState.enlargeDirection && !myState.dragging) {
-                return;
+            var mx = mouse.x;
+            var my = mouse.y;
+            var boxes = myState.boxes;
+            var l = boxes.length;
+            var border = false;
+            var box = null;
+
+            /* Checks if mouse is hovering over a border. */
+            for (var i = l - 1; i >= 0; i--) {
+                box = boxes[i]; 
+                border = boxes[i].withinBorder(mx, my);
+                if (border) {
+                    break;
+                }
+            }
+
+
+            if (!myState.enlargeDirection && !myState.dragging && !border) {
+                document.body.style.cursor = 'default';
+            } else if (!myState.enlargeDirection && !myState.dragging && border) {
+                myState.doubleArrow(box, mx, my);
```

Comment:
I tried doing that initially, but I realized that the border doesn't necessarily tell you where your mouse is on a box in terms of top, bottom, left, and right. Instead it returns a specific border, say the "top" border, but depending on how you dragged to create your box, that border might actually be the bottom. So I had to create additional conditionals to account for this so that the arrow is always in the correct style.